extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/drive/DriveImporter.java

```diff
@@ -0,0 +1,148 @@
+import org.datatransferproject.datatransfer.google.common.GoogleStaticObjects;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.spi.transfer.provider.ImportResult;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.transfer.models.blob.BlobbyStorageContainerResource;
+import org.datatransferproject.types.transfer.models.blob.DigitalDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DriveImporter implements
+    Importer<TokensAndUrlAuthData, BlobbyStorageContainerResource> {
+  private static final Logger logger = LoggerFactory.getLogger(DriveExporter.class);
+
+  private final GoogleCredentialFactory credentialFactory;
+  private volatile Drive driveInterface;
+  private final JobStore jobStore;
+
+  public DriveImporter(GoogleCredentialFactory credentialFactory, JobStore jobStore) {
+    this(credentialFactory,
+        // Lazily initialized later on
+        null,
+        jobStore);
+  }
+
+  @VisibleForTesting
+  DriveImporter(
+      GoogleCredentialFactory credentialFactory,
+      Drive driveInterface,
+      JobStore jobStore) {
+    this.credentialFactory = credentialFactory;
+    this.driveInterface = driveInterface;
+    this.jobStore = checkNotNull(jobStore, "Job store can't be null");
+  }
+
+  @Override
+  public ImportResult importItem(UUID jobId, TokensAndUrlAuthData authData,
+      BlobbyStorageContainerResource data) throws Exception {
+    String parentId = null;
+    Drive driveInterface = getDriveInterface(authData);
+
+    // Let the parent ID be empty for the root level
+    if (Strings.isNullOrEmpty(data.getId()) || "root".equals(data.getId())) {
+      parentId = importSingleFolder(
+          jobId,
+          driveInterface,
+          "MigratedContent",
+          "root",
+          null);
+    } else {
+      DriveFolderMapping mapping = jobStore.findData(jobId, data.getId(), DriveFolderMapping.class);
+      checkNotNull(mapping, "No mapping found for %s", data.getId());
+      logger.info("Got parent id {} for old Id {} named: {}", parentId, data.getId(), data.getName());
+      parentId = mapping.getNewId();
+    }
+
+    // Uploads album metadata
+    if (data.getFolders() != null && data.getFolders().size() > 0) {
+      for (BlobbyStorageContainerResource folder : data.getFolders()) {
+        importSingleFolder(jobId, driveInterface, folder.getName(), folder.getId(), parentId);
+      }
+    }
+
+    // Uploads photos
+    if (data.getFiles() != null && data.getFiles().size() > 0) {
+      for (DigitalDocument file : data.getFiles()) {
+        importSingleFile(jobId, driveInterface, file, parentId);
+      }
+    }
+
+    return ImportResult.OK;
+  }
+
+  private String importSingleFolder(
+      UUID jobId,
+      Drive driveInterface,
```

Comment:
we don't need to pass in driveInterface since it's a member var, right?

Reply:
You shouldn't access driveInterface directly as its lazy inited, so it should be called via getDriveInterface but then you would have to pass auth data down.  So it seems cleaner to just pass  driveInterface down.

For the current implementation driveInterface will always be set when this function is called, but that seems like a bad assumption to bake in.

I added a comment to the member variable to not access it directly.